### PR TITLE
Migrate CI to GitHub Actions

### LINKED ISSUE #391: Migrate CI to GitHub Actions
Due to changes in the Travis CI billing, the Dask org is migrating CI to GitHub Actions.

This repo contains a `.travis.yml` file which needs to be replaced with an equivalent `.github/workflows/ci.yml` file.

See dask/community#107 for more details.

### PR DESCRIPTION
**Pull request by jacobtomlinson**
Closes #391 

[CI running on fork](https://github.com/jacobtomlinson/s3fs/actions). Will run on future PRs once merged.

### CODE REVIEW

**Comment by martindurant** [comment]
The version string during running of the test comes out as "'0+untagged.1.g638f18a'", which fails a version check requiring s3fs to be recent enough for fsspec. Can you push some tags to your fork, please, and rerun?

**Comment by martindurant** [comment]
(I tried to push a tag, but was denied)

**Comment by jacobtomlinson** [comment]
Thanks @martindurant. Strange you couldn't push as I have the allow edits box checked. Perhaps it's because it's tags and not to the branch.

I've pushed tags and rerun CI.

**Comment by martindurant** [comment]
> Perhaps it's because it's tags and not to the branch.

Exactly

**Comment by martindurant** [comment]
Still says '0+untagged.1.g99267b1' in the build :|

**Comment by jacobtomlinson** [comment]
Hrm strange.

**Comment by martindurant** [comment]
OK, merging this, and then I can actually fix the linting problems.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -49,5 +49,5 @@ jobs:
       - name: Lint
         shell: bash -l {0}
         run: |
-          flake8 gcsfs
-          black gcsfs --check
+          flake8 s3fs
+          black s3fs --check
```
Comment (jacobtomlinson) [inline .github/workflows/ci.yml:53]: Ha! Oops

**Comment by jacobtomlinson** [comment]
Thanks for fixing this @martindurant

**Pull request merged**